luaui/RmlWidgets/gui_awards/gui_awards.lua

```diff
@@ -0,0 +1,338 @@
+function widget:GetInfo()
+	return {
+		name = "Awards",
+		desc = "UI with awards after game ends",
+		author = "Floris (original: Bluestone)",
+		date = "July 2021",
+		license = "GNU GPL, v2 or later",
+		layer = -3,
+		enabled = true,
+	}
+end
+
+local document
+widget.rmlContext = nil
+
+local playerListByTeam = {}
+local isVisible = false
+local documentModelHandle
+local chobbyLoaded = (Spring.GetMenuName and string.find(string.lower(Spring.GetMenuName()), "chobby") ~= nil)
+
+local injectableRml = {
+	player = "<span class=\"other-awards-leader\" style=\"color: %s;\">%s</span>",
+	score = "<span class=\"other-awards-score\">%s</span>"
+}
+
+local modelForPresenter = {
+	resourcesDestroyed = {
+		visible = false,
+		size = 3,
+		firstLeader = {
+			playerName = "Test",
+			score = -1,
+			playerColor = "rgb(255,0,0)",
+		},
+		otherLeadersOrdered = {
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+		},
+	},
+	enemiesDestroyed = {
+		visible = false,
+		size = 3,
+		firstLeader = {
+			playerName = "Test",
+			score = -1,
+			playerColor = "rgb(255,0,0)",
+		},
+		otherLeadersOrdered = {
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+		},
+	},
+	resourcesEfficiency = {
+		visible = false,
+		size = 3,
+		firstLeader = {
+			playerName = "Test",
+			score = -1,
+			playerColor = "rgb(255,0,0)",
+		},
+		otherLeadersOrdered = {
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+		},
+	},
+	traitor = {
+		visible = false,
+		size = 3,
+		firstLeader = {
+			playerName = "Test",
+			score = -1,
+			playerColor = "rgb(255,0,0)",
+		},
+		otherLeadersOrdered = {
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+			{
+				playerName = "Test",
+				score = -1,
+				playerColor = "rgb(255,0,0)",
+			},
+		},
+	},
+	goldenCow = {
+		visible = false,
+		leader = {
+			playerName = "Test",
+			playerColor = "rgb(255,0,0)",
+		},
+	},
+	otherAwards = {},
+}
+
+local function leave()
+	if chobbyLoaded then
+		Spring.Reload("")
+	else
+		Spring.SendCommands("quitforce")
+	end
+end
+
+local function showGraph()
+	document:Close()
+	Spring.SendCommands("endgraph 2")
+end
+
+local function initializeRmlContext()
+	widget.rmlContext = RmlUi.CreateContext(widget.whInfo.name)
```

Comment:
Why are you creating a new context? Use the 'shared' context. Each widget gets their own document in shared context, not their own context.